runtime/ftplugin/sh.vim

```diff
@@ -41,6 +41,19 @@ if (has("gui_win32") || has("gui_gtk")) && !exists("b:browsefilter")
   let b:undo_ftplugin ..= " | unlet! b:browsefilter"
 endif
 
+if (exists('b:is_bash') && (b:is_bash == 1)) ||
+      \ (exists('b:is_sh') && (b:is_sh == 1))
+  if !has('gui_running') && executable('less')
+    command! -buffer -nargs=1 Help silent exe '!bash -c "{ help "<args>" 2>/dev/null || man "<args>"; } | LESS= less"' | redraw!
+  elseif has('terminal')
+    command! -buffer -nargs=1 Help silent exe ':term bash -c "help "<args>" 2>/dev/null || man "<args>""'
+  else
+    command! -buffer -nargs=1 Help echo system('bash -c "help <args>" 2>/dev/null || man "<args>"')
+  endif
```

Comment:
Why is `bash` always used here even if `b:is_bash` is not true? On Debian/Ubuntu, `sh` is symlinked to `dash` but not to `bash`.

For example,
```sh
#!/bin/sh
type time
```

prints `time is a shell keyword` on a machine where `sh` is `bash`, e.g. RHEL; on Debian/Ubuntu, it prints `time is /usr/bin/time`  or `time is not found` since `time` is not a dash builtin.
Hence, `K` on `time` must not invoke `bash -c 'help time'` unless it is a bash script.

Reply:
That's a good point; bash was thought to be standard. How about a check if `sh` is symlinked to `bash` if `is_sh` is true? Or rather start from the premise that the user intended not to use `bash` is `is_sh` is set?

Reply:
well, it will be impossible to guess from Vim, whether a certain keyword is shell-builtin or an external command. I think this is rather a best effort approach here and not the ultimate correct solution under all circumstances however.

Reply:
@dkearns might have a more informed opinion. For reference, here's what `:help sh.vim` says:

```
Vim attempts to determine which shell type is in use by specifying that
various filenames are of specific types, e.g.: >

    ksh : .kshrc* *.ksh
    bash: .bashrc* bashrc bash.bashrc .bash_profile* *.bash
<
See $VIMRUNTIME/filetype.vim for the full list of patterns.  If none of these
cases pertain, then the first line of the file is examined (ex. looking for
/bin/sh  /bin/ksh  /bin/bash).  If the first line specifies a shelltype, then
that shelltype is used.  However some files (ex. .profile) are known to be
shell files but the type is not apparent.  Furthermore, on many systems sh is
symbolically linked to "bash" (Linux, Windows+cygwin) or "ksh" (Posix).

One may specify a global default by instantiating one of the following
variables in your <.vimrc>:

   ksh: >
	let g:is_kornshell = 1
<   posix:  (using this is nearly the same as setting g:is_kornshell to 1) >
	let g:is_posix     = 1
<   bash: >
	let g:is_bash	   = 1
<   sh: (default) Bourne shell >
	let g:is_sh	   = 1

<   (dash users should use posix)

If there's no "#! ..." line, and the user hasn't availed himself/herself of a
default sh.vim syntax setting as just shown, then syntax/sh.vim will assume
the Bourne shell syntax.  No need to quote RFCs or market penetration
statistics in error reports, please -- just select the default version of the
sh your system uses and install the associated "let..." in your <.vimrc>.
```

Reply:
Reading this, Ubuntu users should set `g:is_posix` to resolve the issue. However, as Ubuntu is rather common, and many users don't know about `g:is_posix`, one could think about making detecting `dash`.

Reply:
I prefer the procedure of https://github.com/vim/vim/pull/5566#discussion_r1322565002 to the realpath of sh.

Reply:
If you wish, you can add a pull request for a check for whether `is_sh` or `is_posix` is true, `resolve('\bin\sh')` points to `dash` and then supply the `dash` variant of `bash -c "help <args>" 2>/dev/null || man "<args>"`